run_headless_sim.sh

```diff
@@ -0,0 +1,377 @@
+    addRover ${ROVER_NAMES[i]} ${ROVER_POSITIONS_X[i]} ${ROVER_POSITIONS_Y[i]} 0 0 0 ${ROVER_YAWS[i]}
+    sleep $MODEL_ADD_INTERVAL
+    startRoverNodes ${ROVER_NAMES[i]}
+done
+
+echo "Finished adding rovers."
+
+#---------------------------------------------------------#
+
+sleep $MODEL_ADD_INTERVAL
+
+echo "Setting rovers to autonomous mode..."
+
+# Send the autonomous command to all rovers
+for (( i=0;i<$NUM_ROVERS;i++ ));
+do
+    # Publish the autonomous mode command ("2") to each rover. Latch the message ("-l").
+    rostopic pub -l /${ROVER_NAMES[i]}/mode std_msgs/UInt8 2 &
+    echo "Publishing 2 on /${ROVER_NAMES[i]}/mode"
+done
+ 
+echo "Finished setting rovers to autonomous mode."
+
+# Read output file path from command line
+SCORE_OUTPUT_PATH=$3
+
+mkdir -p $(dirname $SCORE_OUTPUT_PATH)
+
+echo "User specified $SCORE_OUTPUT_PATH as the file to which score information should be appended."
+
+#---------------------------------------------------------#
+
+# Read the experiment time from command line
+EXPERIMENT_DURATION_IN_MINUTES=$4
+EXPERIMENT_DURATION_IN_SECONDS=$(( $EXPERIMENT_DURATION_IN_MINUTES*60 ))
+
+echo "Experiment duration will be $EXPERIMENT_DURATION_IN_SECONDS seconds."
+
+# Read the current sim time (in seconds) from the ros topic /clock
+# Uses awk to match on first occurence of "secs: number".
+# {print $2} prints the number to the shell variable 
+# exit the awk after first match because we dont care about "nsecs: number"
+CURRENT_TIME=$(rostopic echo -n 1 /clock | awk '/secs: [0-9]+$/{print $2; exit}')
+START_TIME=$(rostopic echo -n 1 /clock | awk '/secs: [0-9]+$/{print $2; exit}')
+
+echo "Initialised current gazebo time to $CURRENT_TIME"
+echo "Initialised start gazebo time to $START_TIME"
+
+# Let the simulation run until the experiment duration is reached
+YELLOW='\033[0;33m'
+PURPLE='\033[0;35m'
+NC='\033[0m' # No Color
+
+EXPERIMENT_REAL_SETUP_END_TIME_IN_SECONDS=$(date +%s)
+EXPERIMENT_REAL_START_TIME_IN_SECONDS=$(date +%s)
+
+#Collect the score over time and write to screen and file
+echo "Time (s), Score\n" >> $SCORE_OUTPUT_PATH
+
+until (( $CURRENT_TIME-$START_TIME>=$EXPERIMENT_DURATION_IN_SECONDS )); do
+    # Update the current sim time
+    CURRENT_TIME=$(rostopic echo -n 1 /clock | awk '/secs: [0-9]+$/{print $2; exit}')
+    echo -e "Experiment time remaining ${YELLOW}$(( $EXPERIMENT_DURATION_IN_SECONDS-($CURRENT_TIME-$START_TIME) ))${NC} seconds."
+    # Pipe /score ros topic to output file
+    echo -e "${PURPLE}Time: $(($CURRENT_TIME-$START_TIME)), Score: $(rostopic echo -n 1 /collectionZone/score | sed 's/[^0-9]*//g')${NC}"
+    echo "$(($CURRENT_TIME-$START_TIME)), $(rostopic echo -n 1 /collectionZone/score | sed 's/[^0-9]*//g')\n" >> $SCORE_OUTPUT_PATH
+    
+    sleep $END_EXPERIMENT_CHECK_INTERVAL
+done
+
+echo "The specified experiment duration ($EXPERIMENT_DURATION_IN_MINUTES) has elapsed. End autonomous mode for all rovers."
+
+# Send the autonomous command to all rovers
```

Comment:
This should be sending the manual command or some stop command to all rovers, rather than the autonomous command again.